core/templates/dev/head/filtersSpec.js

```diff
@@ -264,4 +265,84 @@ describe('Testing filters', function() {
+    'This set of tabs shows some hints.' +
+    ' Click on the other tabs to display the relevant hints.&amp;quot;},' +
+    '{&amp;quot;title&amp;quot;:&amp;quot;Hint 1&amp;quot;,' +
+    '&amp;quot;content&amp;quot;:&amp;quot;This is a first hint.&amp;quot;},' +
+    '{&amp;quot;title&amp;quot;:&amp;quot;Hint 2&amp;quot;,' +
+    '&amp;quot;content&amp;quot;:&amp;quot;&amp;lt;p&amp;gt;' +
+    'Stuff and things&amp;lt;/p&amp;gt;&amp;quot;}]" ' +
+    'exploration-id-with-value="&amp;quot');
+    var OPPIA_IMG = ('<img src="image.png" ' +
+    'class="oppia-noninteractive-image block-element" ' +
+    'alt-with-value="&amp;quot;&amp;quot;" ' +
+    'caption-with-value="&amp;quot;&amp;quot;" ' +
+    'filepath-with-value="&amp;quot;DearIDPodcast_sm.png&amp;quot;" ' +
+    'exploration-id-with-value="&amp;quot;5WWM85JODA2X&amp;quot;">');
+    var OPPIA_VIDEO = ('<img ' +
+    'src="https://img.youtube.com/vi/JcPwIQ6GCj8/hqdefault.jpg" ' +
+    'class="oppia-noninteractive-video block-element" ' +
+    'video_id-with-value="" start-with-value="0" end-with-value="0" ' +
+    'autoplay-with-value="false" exploration-id-with-value="">');
+    var IMG_HTML = ('<a ' +
+    'href="https://en.wikipedia.org/wiki/File:The_Purloined_Letter.jpg" ' +
+    'class="image"><img alt="The Purloined Letter.jpg" ' +
+    'src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/65/' +
+    'The_Purloined_Letter.jpg/220px-The_Purloined_Letter.jpg" width="220" ' +
+    'height="178"></a>');
+    var OTHER_TAG_LINK = ('<a href=""><img src="linkimage.jpg" ' +
+    'class="other-tag"></a>');
+    var INVALID_TAG_LINK = ('<a href="example.com" class="invalid-tag"></a>');
+    var DANGEROUS_SCRIPT_IMG = ('<img src="w3javascript.gif" ' +
+    'onload="loadImage()" width="100" height="132">');
+    var DANGEROUS_NESTED_SCRIPT = ('<scr<script>ipt>alert(42);' +
+    '</scr</script>ipt>');
+    var whitelistedTags = [
+      'oppia-noninteractive-collapsible',
+      'oppia-noninteractive-image',
+      'oppia-noninteractive-link',
+      'oppia-noninteractive-math',
+      'oppia-noninteractive-tabs',
+      'oppia-noninteractive-video',
+      'other-tag'
+    ];
+
+    expect(
+      $filter('stripFormatting')(LINK_HTML, whitelistedTags)
+    ).toEqual('1800');
+    expect(
+      $filter('stripFormatting')(IMG_HTML, whitelistedTags)
+    ).toEqual('');
+    expect(
+      $filter('stripFormatting')(OPPIA_TABS, whitelistedTags)
+    ).toEqual(OPPIA_TABS);
+    expect(
+      $filter('stripFormatting')(OPPIA_IMG, whitelistedTags)
+    ).toEqual(OPPIA_IMG);
+    expect(
+      $filter('stripFormatting')(OPPIA_VIDEO, whitelistedTags)
+    ).toEqual(OPPIA_VIDEO);
+    expect(
+      $filter('stripFormatting')(DANGEROUS_SCRIPT_IMG, whitelistedTags)
+    ).toEqual('');
+    expect(
+      $filter('stripFormatting')(OTHER_TAG_LINK, whitelistedTags)
+    ).toEqual('<a href=""><img src="linkimage.jpg" class="other-tag">');
```

Comment:
Oh, this is very interesting -- and it's something I didn't realize until I looked at the tests. (This is an example of how good tests can help illuminate the behavior of the code.)

It turns out that the components are not oppia-noninteractive-* tags, as I had thought from looking at the implementation of the filter. Instead, they are img tags with particular class names.

I think that the regex needs to look for that instead -- i.e., it should explicitly match img tags that have particular classnames. So it should exclude tags that start with `<img` and that have `class="` somewhere after that and then, somewhere after that, `{{whitelisted-class-name}}` before the closing quotation mark corresponding to the opening one for 'class', and finally `>`.